lib/src/robot_interface/robot_interface.cpp

```diff
@@ -327,11 +336,41 @@ void RobotInterface::IRCb(const sensor_msgs::RangeConstPtr& msg)
 
 void RobotInterface::filterForces()
 {
-    _filt_force[0] = (1 - FORCE_ALPHA) * _filt_force[0] + FORCE_ALPHA * _curr_wrench.force.x;
-    _filt_force[1] = (1 - FORCE_ALPHA) * _filt_force[1] + FORCE_ALPHA * _curr_wrench.force.y;
-    _filt_force[2] = (1 - FORCE_ALPHA) * _filt_force[2] + FORCE_ALPHA * _curr_wrench.force.z;
+    double delta_time = ros::Time::now().toSec() - _filt_updated;
+
+    std::vector<double> new_filt;
+    std::vector<double> predicted_filt;
+
+    // initial attempt to update filter using an EMA
```

Comment:
Is it the initial or the final attempt?
Also, what is EMA? I'm undecided between "Emergency Medical Associates" and "European Music Awards" :smile: https://www.google.com/search?client=ubuntu&channel=fs&q=ema&ie=utf-8&oe=utf-8